platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/DefaultMapProperty.java

```diff
@@ -605,6 +629,34 @@ public ExecutionTimeValue<? extends Map<K, V>> calculateExecutionTimeValue() {
             );
         }
 
+        @Override
+        @Nullable
+        public MapCollector<K, V> getCollectorForKey(K key) {
+            for (MapCollector<K, V> collector : collectors) {
+                ImmutableSet.Builder<K> keysBuilder = ImmutableSet.builder();
+                // We are intentionally ignoring returned Value's side-effects here
```

Comment:
❓ Should we do it?

Reply:
> ❓ Should we do it?

I think it is fine, this isn't a real "read" at this point. Same goes to the other question.